lib/gollum/public/gollum/javascript/gollum.js.erb

```diff
@@ -102,10 +102,17 @@ function preparePage () {
   }
 }
 
-function flashNotice(type, notice) {
+function flashNotice(type, notice, button_label, button_function, button_type) {
   // accepted types: info, success, warn, error
-  html = '<p><div class="flash flash-' + type +'"><button class="flash-close js-flash-close" type="button" onclick="parentNode.remove()"><%=rocticon('x')%></button>' + notice + '</div></p>';
+  nested_button_html = '';
+  if (typeof(button_label) !== 'undefined'  && typeof(button_function) !== 'undefined') {
+    button_type = (typeof(button_type) !== 'undefined' && button_type == 'danger') ? ' btn-danger' : '';
+    nested_button_html = '<span class="px-2"><button type="button" class="btn btn-sm' + button_type + '" onclick="' + button_function + '">' + button_label + '</button></span>'
+  }
+  html = '<p><div id="gollum-flash" class="flash flash-' + type +'"><button class="flash-close js-flash-close" type="button" onclick="parentNode.remove()"><%=rocticon('x')%></button>' + notice + nested_button_html + '</div></p>';
+  $('#gollum-flash').remove();
```

Comment:
Don't quite understand how this plays with the timeout below. Doesn't this just remove any earlier flash notices, potentially before the user could see them?

Reply:
I'm especially worried about cases in which, say, a success notice might remove an error or warn notice

Reply:
` $('#gollum-flash').remove();` makes sure that there is no more than one flash notice at a time. So yes, if you would have two notices following each other in quick succession, the first one would be removed in favor of the second, more recent one.

> I'm especially worried about cases in which, say, a success notice might remove an error or warn notice

Here's the scenario in favor of this behavior. You upload a file, which turns out to already exist. You get an error. You upload your file again, under another name. This time the upload succeeds. The intuitive behavior, I think, is for the more recent success flash message to supersede the earlier error message.

Reply:
Using one at a time is also in line with the Primer CSS style guide:

![Screen Shot 2019-10-09 at 17 32 44](https://user-images.githubusercontent.com/571173/66496206-d7644480-eaba-11e9-9542-9fbb5ca9c123.png)

Reply:
OK :)

Reply:
Ok :)